src/System.Management.Automation/engine/AutomationEngine.cs

```diff
@@ -38,27 +38,19 @@ internal AutomationEngine(PSHost hostInterface, InitialSessionState iss)
 #if !UNIX
             // Update the env variable PathEXT to contain .CPL
             var pathext = Environment.GetEnvironmentVariable("PathEXT");
-            pathext = pathext ?? string.Empty;
-            bool cplExist = false;
-            if (pathext != string.Empty)
+
+            if (string.IsNullOrEmpty(pathext))
             {
-                string[] entries = pathext.Split(Utils.Separators.Semicolon);
-                foreach (string entry in entries)
-                {
-                    string ext = entry.Trim();
-                    if (ext.Equals(".CPL", StringComparison.OrdinalIgnoreCase))
-                    {
-                        cplExist = true;
-                        break;
-                    }
-                }
+                Environment.SetEnvironmentVariable("PathEXT", ".CPL");
             }
-
-            if (!cplExist)
+            else if (!(pathext.EndsWith(";.CPL", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
It could end with `;.CPL;`, or even multiple `;` at end.
I think you want `pathext.TrimEnd(';')` first.

Reply:
I think we can formally ignore this. By default Windows set correct value. Formally user or another application can set empty value. But we explicitly ignore such empty values in InitPAthExtCache() method.